Add support for IPv6 on Juniper devices

bdlamprecht opened the pull request:
I needed to get YANG support for IPv6 addresses to work on Juniper devices, so I simply copied the existing `ipv4.yaml` and altered it as needed.

This involved adding this section under the main `config` stanza:
```
        dup-addr-detect-transmits:
            _process: not_implemented
```

Also, I discovered that the interface name `em` was not included in the `openconfig-interfaces/interfaces.yaml` file so I added that too.

bdlamprecht committed:
Add 'em' to list of interfaces for 'junos'

Commit:
Adding an 'ipv6.yaml' file for 'junos'

bdlamprecht commented:
I messed something up again so the `travis-ci` checks fail but don't know what I did.

If someone could help me understand why it is failing, I'd be happy to fix it.

ckishimo commented:
@bdlamprecht I think you need to add the following ipv6 block just below the ipv4 for each interface and subinterface in the `candidate.json` and `expected.json` in the test files that are failing

```
                   "ipv6": {
                        "config": {
                            "enabled": false
                        }
                    }
```
BTW, the `em` interface was already present in`interfaces.yaml`

Commit:
Attempting to fix tests for failing Travis-CI tests per suggestion form @ckishimo
Specifically, this is for 'test_parse[junos-config-openconfig_interfaces-default]'.

bdlamprecht commented:
This _appears_ to have solved the 1st failed test, but seems to have caused a continuous failing loop for this test:

```
test/integration/test_tutorial.py::Tests::test_populating_from_file[profile1] FAILED
```

Not sure if this needs to me manually stopped or not, but unsure of what else I need to change.

ckishimo commented:
@bdlamprecht I think there is one more file to update
`test/integration/tutorial_data/test_populating_from_file/junos/expected.json`

bdlamprecht committed:
Another attempt at fixing Travis-CI failure per @ckishimo

This time it is to fix the error for 'test/integration/test_tutorial.py::Tests::test_populating_from_file[profile1]'

bdlamprecht commented:
Yeah that appears to have fixed the error with `py3.6`, but there is something wrong with `py2.7`.

Do the developers have any intentions to remove support for Python2 similar to what they've described with `nornir`?

dbarrosop commented:
I wouldn't mind dropping py2.7 support. Currently traveling but let me take a look in the upcoming days, I saw the errors and they are kind of weird.

dbarrosop commented:
I just removed py2.7 support in develop, if you rebase you should be good. However, I'd like to add the translation counterpart to have bidirectional support. Would you mind adding that?

bdlamprecht commented:
I'd be happy to rebase on development.
But I'm not sure what you mean by this statement:
> I'd like to add the translation counterpart to have bidirectional support.

Could you please elaborate on what that means?

dbarrosop commented:
Unfortunately I don't have the cycles to keep maintaining this project so as of now it's officially abandoned. If someone wants to fork the project and maintain it don't hesitate to contact me to add a link to the fork and direct existing and new users to the new fork. However, I'd recommend checking the following alternative:

* [yangify](https://github.com/networktocode/yangify)

Yangify has a cleaner interface that is easier to maintain and happens to be orders of magnitude faster and also more memory efficient.

If you are an existing user of napalm-yang and are worried about this announcement and/or want help migrating to some other project don't hesitate to contact me.
